### PR TITLE
File Explorer: Add tooltips

### PR DESCRIPTION
**Pull request by glennsl**
I'd add a screenshot but, you know, it's a tooltip...

### CODE REVIEW

File: src/Components/Tooltip.re [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      };
+    };
+  };
+
+  let setTooltip = tooltip => debouncedSetTooltip(Some(tooltip));
+  let clearTooltip = () => debouncedSetTooltip(None);
+
+  let%component make = (~theme, ~font, ()) => {
+    let%hook (current, setCurrent) = Hooks.state(None);
+    internalSetTooltip := setCurrent;
```
Comment (glennsl) [inline src/Components/Tooltip.re]: I've used a simpler mechanism here to communicate between the trigger/hotspot/anchor and the overlay components while still triggering a re-render.

In the context menu I used the state to pass information between them and trigger re-render. Here I use a hooks instead, and then assign the hook's setter to a global ref, which can then be called by the trigger component, through some convenience functions to cover over the rough edges. Much more contained, and works pretty well I think.

**Comment by bryphe** [comment]
> I'd add a screenshot but, you know, it's a tooltip...

Maybe I'm easily excited but I think it looks awesome! 😎 

image

This will be very useful to have - I can think of lots of places it would be helpful and aid in discoverability.

File: src/Components/Tooltip.re (hunk 1 above)
Comment (bryphe) [reply src/Components/Tooltip.re]: Neat!

File: src/Components/Tooltip.rei [hunk 2]
```diff
@@ -0,0 +1,16 @@
+open Oni_Core;
+open Revery.UI;
+
+module Overlay: {
```
Comment (bryphe) [inline src/Components/Tooltip.rei]: I like the API you designed for this (and using the `Trigger` module, but `included`, so the interface is streamlined) - seems like it'll be very easy to add in other spots, too.

**Pull request merged**